jujugui/static/gui/src/app/components/search-results-item/search-results-item.js

```diff
@@ -0,0 +1,238 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2013 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('search-results-item', function(Y) {
+
+  juju.components.SearchResultsItem = React.createClass({
+    /**
+      Generate the element for the special flag.
+
+      @method _generateSpecialFlag
+      @returns {String} The generated elements.
+    */
+    _generateSpecialFlag: function() {
+      var special = this.props.item.special;
+      if (!special) {
+        return;
+      }
+      return (
+          <span className="special-flag"></span>
+      );
+    },
+
+    /**
+      Generate the elements for the tag list.
+
+      @method _generateTagList
+      @returns {String} The generated elements.
+    */
+    _generateTagList: function() {
+      var components = [];
+      var tags = this.props.item.tags || [];
+      if (tags.length === 0) {
+        return <span>&nbsp;</span>;
+      }
+      tags.forEach(function(tag) {
+        components.push(
+          <li className="tag-list--item"
+            key={tag}
+            role="button" tabIndex="0"
+            onClick={this._handleTagClick.bind(this, tag)}>
+            {tag}
+          </li>
+        );
+      }, this);
+      return components;
+    },
+
+    /**
+      Generate the elements for the series list.
+
+      @method _generateSeriesList
+      @returns {String} The generated elements.
+    */
+    _generateSeriesList: function() {
+      var series = this.props.item.series;
+      var components = [];
+      if (series.length === 0) {
+        return <span>&nbsp;</span>;
```

Comment:
Does React actually like `&nbsp`? I was having some issues elsewhere trying to use it.